### ISSUE TITLE
Work spreading idea: Buckets

### ISSUE THREAD

**Issue by Mengmoshu** [Mengmoshu/Resource_Relocation_Machines#14]
Create a number of "buckets" equal to the number of ticks per cycle of a machine set. Evenly spread machines across the buckets. One bucket gets run per tick.

Concerns:
* Ensuring suitable uniformity as machines are removed from buckets.
* * Removals trigger a re-ranking of the buckets?
* * Insertions trigger a re-ranking of the buckets?
* What to do with different machine tick rates.

**Comment by Mengmoshu** [comment]
A bit of code Nexela has shared from Nanobots for reference.

```lua
local function poll_players(event)
    --Run logic for nanobots and power armor modules
    --if event.tick % math.ceil(#game.connected_players/cfg.poll_rate) == 0 then
    if event.tick % max(1, floor(cfg.poll_rate/#game.connected_players)) == 0 then
        local last_player, player = next(game.connected_players, global._last_player)
        --Establish connected, non afk, player character
if player and is_connected_player_ready(player) then
```

**Comment by Mengmoshu** [comment]
Closed by PR #19
